### PR TITLE
Add more logging and proper permissions to image scan

### PR DESCRIPTION
**Pull request by etsauer**
@JaredBurck @sabre1041 @pabrahamsson

### CODE REVIEW

**Comment by Gl4di4torRr** [comment]
@etsauer nice! For those reviewing, I ran a the image scanner and it took a couple of hours. Not knowing the percentage complete is annoying when you have a large amount of images being scanned.

**Comment by etsauer** [comment]
@bbeaudoin see updates.

File: image-scanning/templates/pipeline.yml [hunk 1]
```diff
@@ -98,15 +99,23 @@ objects:
...
+                    SA_FULLNAME=\$(oc whoami)
+                    SA_NAME="\${SA_FULLNAME##*:}"
+                    NAMESPACE=\$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+                    DOCKERCFG_SECRET_NAME=\$(oc get sa -n \$NAMESPACE \$SA_NAME -o jsonpath='{range .secrets[*]}{.name}{"\\n"}{end}' | grep dockercfg)
+
+                    oc get secret -n \$NAMESPACE \$DOCKERCFG_SECRET_NAME -o jsonpath='{.data.\\.dockercfg}' | base64 --decode > \$DOCKER_CFG_HOME/.dockercfg
 
                     { # try
-                      image-inspector --path=\${IMAGE_CONTENT} --scan-results-dir=\${RESULTS} --scan-type=openscap --openscap-html-report --image=\$IMAGE_TO_SCAN
+                      image-inspector --path=\${IMAGE_CONTENT} --scan-results-dir=\${RESULTS} --scan-type=openscap --openscap-html-report --image=\$IMAGE_TO_SCAN --dockercfg=\$DOCKER_CFG_HOME/.dockercfg
```
[the remark this answers is not in the document]
Comment (etsauer) [reply image-scanning/templates/pipeline.yml]: fixed

File: image-scanning/templates/pipeline.yml [hunk 2]
```diff
@@ -112,7 +112,7 @@ objects:
                     NAMESPACE=\$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
                     DOCKERCFG_SECRET_NAME=\$(oc get sa -n \$NAMESPACE \$SA_NAME -o jsonpath='{range .secrets[*]}{.name}{"\\n"}{end}' | grep dockercfg)
 
-                    oc get secret -n \$NAMESPACE \$DOCKERCFG_SECRET_NAME -o jsonpath='{.data.\\.dockercfg}' | base64 --decode > \$DOCKER_CFG_HOME/.dockercfg
+                    oc get secret -n \$NAMESPACE \$DOCKERCFG_SECRET_NAME -o jsonpath='{.data.\\.dockercfg}' | base64 --decode > \$DOCKER_CFG_HOME/config.json
```
[the remark this answers is not in the document]
Comment (etsauer) [reply image-scanning/templates/pipeline.yml]: This script is just grabbing a secret that is generated from the creation of a service account. Not control over the fields created or type of config.

**Comment by etsauer** [comment]
@bbeaudoin when I look at line 115 and 118, they match. are seeing something different?

**Comment by sabre1041** [comment]
@etsauer Running through this resulted in partial success

1. Docker configuration raised by @bbeaudoin appears to be set up properly as images from the registry is pulled
2. Images from the Red Hat registry cannot be retrieved as the cluster that I am using leverages images from registry.redhat.io which requires authentication. This would require a subsequent enhancement

**Pull request merged**